weave/trace/weave_client.py

```diff
@@ -961,32 +982,136 @@ def query_costs(
         )
         return res.results
 
-    ################ Internal Helpers ################
-
-    def _ref_is_own(self, ref: Ref) -> bool:
-        return isinstance(ref, Ref)
-
-    def _project_id(self) -> str:
-        return f"{self.entity}/{self.project}"
+    ################# Object Saving ##################
 
-    # This is used by tests and op_execute still, but the save() interface
-    # is nicer for clients I think?
     @trace_sentry.global_trace_sentry.watch()
     def _save_object(self, val: Any, name: str, branch: str = "latest") -> ObjectRef:
-        self._save_nested_objects(val, name=name)
+        """Save an object to the weave server and return it's Ref. This is the top
+        level entry point for saving any data to the weave server. Importantly, it
+        will also save all children objects that are "Refable".
+
+        Args:
+            val: The object to save.
+            name: The name to save the object under.
+            branch: The branch to save the object under. Defaults to "latest".
+
+        Returns:
+            An ObjectRef to the saved object.
+        """
+        if isinstance(val, WeaveTable):
```

Comment:
I added these no-op conditions for future reference, but did not want to introduce logical changes in this PR